### PR TITLE
update usage of logging exporter

### PR DESCRIPTION
**Pull request by codeboten**
The logging exporter is being deprecated in favour of the debug exporter. This changes references to the logging exporter to use the new debug exporter instead.

Note: this will remain in draft until v0.86.0 is released which will be the first release to include the debug exporter.

### CODE REVIEW

File: README.md
Comment (jaronoff97) [reply README.md:51]: yeah we should probably changelog this

Comment (codeboten) [reply README.md:51]: Added changelog

**Comment by codeboten** [comment]
For the operator, should both the logging and debug exporter be available by default?

**Comment by pavolloffay** [comment]
@codeboten you can update https://github.com/open-telemetry/opentelemetry-operator/blob/main/versions.txt#L5 to 0.86 which will use the latest collector. It should fix failing e2e tests.

File: README.md
Comment (pavolloffay) [reply README.md:51]: The PR changes just docs and e2e tests. Not sure why the changelog is needed.  Users defining the logging exporter in their manifests will need to migrate explicitly. 

@codeboten is the collector printing a warning if the logging exporter is used? The operator could parse the config and print a warning in the validating webhook.

Comment (codeboten) [reply README.md:51]: This deprecation warning appears for any components that are marked as deprecated

**Comment by codeboten** [comment]
@pavolloffay rebased and bumped the version. Looks like there's still some failure. Will try to dig into them further

**Comment by codeboten** [comment]
@pavolloffay @jaronoff97 finally got around to fixing the tests. PTAL

File: README.md
Comment (jaronoff97) [reply README.md:51]: this sounds like a good followup task, but something we can do later

File: versions.txt
Comment (codeboten) [reply versions.txt:5]: I'm not sure I can, since v0.86.0 is the first version to include the debug exporter. if it's preferable, i'm ok waiting for the new operator to release before merging this PR

Comment (pavolloffay) [reply versions.txt:5]: It's fine to keep it here, the collector version can be bumped before the release (e.g. to implement upgrade procedures)

Comment (pavolloffay) [reply versions.txt:5]: Though, not sure if our upgrade test can handle that, The job failed I have restarted it

Comment (jaronoff97) [reply versions.txt:5]: new release is in, so this should be okay. @codeboten can you rebase on main?

**Review by jaronoff97 — APPROVED** [review]
this LGTM, I'm going to merge after the tests pass

**Pull request merged**
